okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.java

```diff
@@ -219,6 +220,10 @@ private boolean recover(IOException e, boolean routeException, Request userReque
     return true;
   }
 
+  private boolean isBodyUnrepeatable(IOException e, RequestBody body) {
```

Comment:
I’m a little reluctant to put these two checks into a single method. The method name suggests that it has nothing to do with which exception was thrown.

I think the goal is to detect that request sending never started, which is good. Maybe we could do that by renaming `routeException` in the calling code to `requestSendStarted` ? Previously that was 1:1 with route exceptions and now we’ve got this extra case that seems similar.